sycl/test-e2e/Graph/Update/FreeFunctionKernels/dyn_work_group_memory_basic.cpp

```diff
@@ -0,0 +1,65 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+// Extra run to check for immediate-command-list in Level Zero
+// RUN: %if level_zero %{env SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=1 %{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+//
+// XFAIL: cuda
+// XFAIL-TRACKER: https://github.com/intel/llvm/issues/16004
+
+// Tests updating dynamic_work_group_memory with a new size.
+
+#include "../../graph_common.hpp"
+#include "free_function_kernels.hpp"
+
+int main() {
+  constexpr int LocalSize{32};
+  nd_range<1> NDRange{Size, LocalSize};
+
+  queue Queue{};
+  context Ctxt{Queue.get_context()};
+
+  int *PtrA = malloc_device<int>(Size, Queue);
+  std::vector<int> HostDataA(Size);
+
+  exp_ext::command_graph Graph{Ctxt, Queue.get_device()};
+  exp_ext::dynamic_work_group_memory<int[]> DynLocalMem{Graph, LocalSize};
+
+  Queue.memset(PtrA, 0, Size * sizeof(int)).wait();
+
+#ifndef __SYCL_DEVICE_ONLY__
```

Comment:
This ifdef may not longer be required if this PR is based off of tip, see https://github.com/intel/llvm/pull/17398

Reply:
oh cool. Done